.github/workflows/build_and_run_chain_simulator_and_execute_system_test.yml

```diff
@@ -0,0 +1,341 @@
+name: Build and Smoke Test
+
+on:
+  pull_request:
+    branches:
+      - 'main'
+      - 'master'
+      - 'rc/*'
+  workflow_dispatch:
+  issue_comment:
+    types: [created]
+
+permissions:
+  issues: write
+  pull-requests: write
+  contents: read
+
+jobs:
+  build-and-test:
+    if: |
+      github.event_name == 'pull_request' ||
+      (github.event_name == 'issue_comment' && contains(github.event.comment.body, 'Run Tests:')) ||
+      github.event_name == 'workflow_dispatch'
+
+    strategy:
+      matrix:
+        #TODO Include Macos support later on 
+        runs-on: [ubuntu-latest]
+    runs-on: ${{ matrix.runs-on }}
+    env:
+      BRANCH_NAME: ${{ github.head_ref || github.ref_name }}
+      TARGET_BRANCH: ""
+      MX_CHAIN_GO_TARGET_BRANCH: ""
+      MX_CHAIN_SIMULATOR_TARGET_BRANCH: ""
+      MX_CHAIN_TESTING_SUITE_TARGET_BRANCH: ""
+
+    steps:
+      - name: Fetch Latest Comment
+        if: github.event_name != 'issue_comment'
+        uses: actions/github-script@v7
+        id: fetch_comment
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: |
+            const comments = await github.rest.issues.listComments({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              issue_number: context.issue.number,
+            });
+
+            // Filter for comments containing "Run Tests:"
+            const latestComment = comments.data.reverse().find(comment => comment.body.includes('Run Tests:'));
+
+            if (latestComment) {
+              core.setOutput('latest_comment', latestComment.body);
+            } else {
+              core.setOutput('latest_comment', '');
+            }
+        env:
+          LATEST_COMMENT: ${{ steps.fetch_comment.outputs.latest_comment }}
+
+      - name: Parse Comment for Branches
+        run: |
+          # Use fetched comment if available, otherwise use current event comment
+          COMMENT="${{ steps.fetch_comment.outputs.latest_comment || github.event.comment.body }}"
+
+          # Debug print the comment being used
+          echo "Comment used for parsing: $COMMENT"
+
+          # Extract branch names from the comment
+          if echo "$COMMENT" | grep -q "mx-chain-simulator-go:"; then
+            SIMULATOR_BRANCH=$(echo "$COMMENT" | grep "mx-chain-simulator-go:" | awk -F': ' '{print $2}')
+            echo "MX_CHAIN_SIMULATOR_TARGET_BRANCH=${SIMULATOR_BRANCH}" >> $GITHUB_ENV
+          fi
+
+          if echo "$COMMENT" | grep -q "mx-chain-testing-suite:"; then
+            TESTING_SUITE_BRANCH=$(echo "$COMMENT" | grep "mx-chain-testing-suite:" | awk -F': ' '{print $2}')
+            echo "MX_CHAIN_TESTING_SUITE_TARGET_BRANCH=${TESTING_SUITE_BRANCH}" >> $GITHUB_ENV
+          fi
```

Comment:
This looks great, but wouldn't it be more simple to use an input that defaults to master for the branch name? Also why is the SIMULATOR_BRANCH mentioned here? Shouldn't it be taken from the `github.ref`? Or in the `workflow_dispatch` case use an input instead. I think it is a more clearer approach than adding logic for parsing comments on PRs.

Reply:
About workflow inputs - GH also provides a nice user interface (a simple form) to provide them. Structured input vs. PR comments.

> specify branches directly in comments

Branches can also be specified in the simple GH form.